src/tests/ftest/util/results_utils.py

```diff
@@ -0,0 +1,340 @@
+            int: total duration of all the tests
+        """
+        return sum(test.time_elapsed for test in self.tests)
+
+    @property
+    def passed(self):
+        """Get the total number of passed tests.
+
+        Returns:
+            int: total number of passed tests
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.PASS)
+
+    @property
+    def warned(self):
+        """Get the total number of warned tests.
+
+        Returns:
+            int: total number of warned tests
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.WARN)
+
+    @property
+    def errors(self):
+        """Get the total number of tests with errors.
+
+        Returns:
+            int: total number of tests with errors
+
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.ERROR)
+
+    @property
+    def interrupted(self):
+        """Get the total number of interrupted tests.
+
+        Returns:
+            int: total number of interrupted tests
+
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.INTERRUPT)
+
+    @property
+    def failed(self):
+        """Get the total number of failed tests.
+
+        Returns:
+            int: total number of failed tests
+
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.FAIL)
+
+    @property
+    def skipped(self):
+        """Get the total number of skipped tests.
+
+        Returns:
+            int: total number of skipped tests
+
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.SKIP)
+
+    @property
+    def cancelled(self):
+        """Get the total number of cancelled tests.
+
+        Returns:
+            int: total number of cancelled tests
+
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.CANCEL)
+
+    @property
+    def completed(self):
+        """Get the total number of completed tests.
+
+        Returns:
+            int: total number of tests that were not skipped or cancelled
+
+        """
+        return self.tests_total - self.skipped - self.cancelled
+
+    @property
+    def succeeded(self):
+        """Get the total number of tests that succeeded.
+
+        Returns:
+            int: total number of tests that passed or warned
+
+        """
+        return self.passed + self.warned
+
+    @property
+    def rate(self):
+        """Get the pass rate for the tests.
+
+        Returns:
+            float: pass rate of all the tests
+
+        """
+        if not self.completed:
+            return 0.0
+        return 100 * (float(self.succeeded) / float(self.completed))
```

Comment:
```suggestion
        return round(100 * (float(self.succeeded) / float(self.completed)), 1)
```
You could round it to first decimal place.

Reply:
The rounding is handled in the reporting.